invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -1100,4 +1104,73 @@ public Void inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFa
             }
         });
     }
+
+    @Override
+    public void transferChildCreditToParent(final Account childAccount, final CallContext context) throws InvoiceApiException {
+        transactionalSqlDao.execute(new EntitySqlDaoTransactionWrapper<Void>() {
```

Comment:
⚠️ All the `InvoiceDao` apis use either  `InternalTenantContext` (read-only) or `InternalCallContext` (read-write), so the second argument `CallContext ` should really be in  `InternalCallContext`. Computation of such a context will require making a choice about whether we use the parent `accountId` or child `accountId`. I would suggest child unless t6here is a good reason not to (and inside the method you might need to create another one for the parent). This should be clearly documented inside the  `InvoiceDao` api.